extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -309,9 +341,74 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
   }
 
   /** @override */
+  buildCallback() {
+    super.buildCallback();
+
+    if (this.isResponsive() && !this.responsiveSizeChangePromise_) {
+      // Create a dummy zero-size element so that attemptChangeSize has
+      // something to show if the resize fails.
+      const dummyOverflowElement = document.createElement('div');
+      dummyOverflowElement.setAttribute('overflow', '1');
+      this.element.appendChild(dummyOverflowElement);
+
+      // Attempt to resize to the correct height.
+      const viewport = viewportForDoc(this.element);
+      this.responsiveSizeChangePromise_ =
+          this.attemptChangeSize(
+              AmpAdNetworkAdsenseImpl.getResponsiveHeightForContext(
+                  viewport.getSize()),
+              undefined);
+    }
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    super.onLayoutMeasure();
+
+    if (this.isResponsive() && !this.responsiveAligned_) {
+      this.responsiveAligned_ = true;
+
+      const layoutBox = this.getLayoutBox();
+
+      // Nudge into the correct horizontal position. NB: this must succeed in
+      // order to align the element correctly, so we use changeSize rather than
+      // attemptChangeSize here. But it doesn't actually change the size (it
+      // just changes the horizontal margins).
+      // TODO(charliereams): This is wrong for RTL.
+      this.element.getResources()./*OK*/changeSize(
+          this.element, undefined, undefined, undefined,
+          {left: -1 * layoutBox.left});
+      setStyle(this.element, 'zIndex', 30);
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    const callback = super.layoutCallback();
+    if (this.responsiveSizeChangePromise_) {
+      return callback.then(this.responsiveSizeChangePromise_);
```

Comment:
You can also use onCreativeRender which is executed after render.  Currently this will wait until load fires for non-AMP creatives (FIE load for AMP creatives)